src/juju/apiHooks.ts

```diff
@@ -0,0 +1,114 @@
+import type {
+  SecretsFilter,
+  ListSecretsArgs,
+} from "@canonical/jujulib/dist/api/facades/secrets/SecretsV2";
+import { useEffect, useCallback } from "react";
+import { useSelector } from "react-redux";
+
+import { connectToModel } from "juju/api";
+import type { ConnectionWithFacades } from "juju/types";
+import {
+  getConfig,
+  getUserPass,
+  getWSControllerURL,
+} from "store/general/selectors";
+import { actions as jujuActions } from "store/juju";
+import { getModelByUUID, getModelUUIDFromList } from "store/juju/selectors";
+import { useAppSelector, useAppDispatch } from "store/store";
+
+export const useModelConnection = (
```

Comment:
Nice addition! I see that we are only using this hook here for the moment. Are we exporting it because we might use it in the future, or only doing so for testing purposes?

Reply:
I'm hoping we can use this for all API calls that currently have the Redux store state or the dispatch method passed as arguments.